### PR TITLE
Scrollbars

### PR DESCRIPTION
**Pull request by mortonjt**
Fixes the scrollbar issues in #435

### CODE REVIEW

File: examples/template.html [hunk 1]
```diff
@@ -124,6 +124,10 @@
             animate();
           });
 
+          $(document).ready(function(){
```
Comment (ElDeveloper) [inline examples/template.html]: How is the callback on line 113 different from this?

File: emperor/support_files/js/controller.js [hunk 2]
```diff
@@ -162,19 +162,24 @@ define([
 
     this.renderer.setSize(plotWidth, this.height);
 
-    if (this.colorController !== undefined){
-      // resize the grid according to the size of the container, since we are
-      // inside the tabs we have to account for that lost space, hence the
-      var tabWidth = this.$plotMenu.width() * this.GRID_SCALE,
-      tabHeight = this.$plotMenu.height() * this.GRID_SCALE;
+    // resize the grid according to the size of the container, since we are
+    // inside the tabs we have to account for that lost space, hence the
```
Comment (ElDeveloper) [inline emperor/support_files/js/controller.js]: I just noticed this comment is a bit off, though it was like this before, would you mind deleting the "hence the" at the end of this line?

**Comment by ElDeveloper** [comment]
Thanks so much @mortonjt, though I'm seeing an odd scrolling in the visibility tab, see the animated gif below:

tab-scroll

Notice how I can go past the metadata selectors are scrolled up, aside from that, I think this is looking good.

I remember we had fixed this in your previous branch, right?

File: examples/template.html (hunk 1 above)
Comment (mortonjt) [inline examples/template.html]: This is something that I've been experimenting with.

`$(document).ready(function()){}` seems to be called when all of the visible elements are loaded.
This actually loads after `$(function()){}` is called.

I am not sure why this is the case - it seems counterintuitive, especially considering that they are supposed to be equivalent (at least according to SO).

I encourage you to test this out.

**Comment by mortonjt** [comment]
No.  We never fixed that problem.

File: examples/template.html (hunk 1 above)
Comment (ElDeveloper) [inline examples/template.html]: Hmmm, should we get rid of all occurrences of `$(function())...` in favor of `document`? at least that way we would guarantee that we only display stuff once we truly are ready to display it.

**Comment by ElDeveloper** [comment]
I seem to recall the problem with the visibility tab was that we somehow could only scroll through n-3 elements of the total n.

**Comment by mortonjt** [comment]
I think @ElDeveloper and I just fixed this problem.  @josenavas @antgonza can you have a look and maybe merge?

File: emperor/support_files/js/controller.js (hunk 2 above)
Comment (antgonza) [inline emperor/support_files/js/controller.js]: This still needs to be addressed, right?

**Comment by antgonza** [comment]
1 comment.

File: emperor/support_files/js/controller.js (hunk 2 above)
Comment (ElDeveloper) [inline emperor/support_files/js/controller.js]: yes!

**Comment by josenavas** [comment]
Looks good once @ElDeveloper 's comments is addressed.

File: examples/template.html (hunk 1 above)
Comment (mortonjt) [inline examples/template.html]: Just tested that out.  If we move everything into document.ready(), it breaks.
```javascript
          $(document).ready(function(){
             init();
             animate();
             ec.resize(window.innerWidth, window.innerHeight);
          });
```
Here's something funny - if I have two declarations of document.ready(), it works again
```javascript
          $(document).ready(function(){
             init();
             animate();
          });

          $(document).ready(function(){
             ec.resize(window.innerWidth, window.innerHeight);
          });
```

**Comment by mortonjt** [comment]
Comments addressed

File: examples/template.html (hunk 1 above)
Comment (ElDeveloper) [inline examples/template.html]: Can we just remove this? Or is that not possible?

Comment (mortonjt) [inline examples/template.html]: Not possible.

**Comment by mortonjt** [comment]
Who wants to merge this :)

**Comment by josenavas** [comment]
Are there :beers: as a prize?

**Comment by mortonjt** [comment]
Blazing fast! +1 :beer: for @josenavas

**Pull request merged**